apps/OboeTester/app/src/main/cpp/cpu/AudioWorkloadTest.h

```diff
@@ -251,12 +277,17 @@ class AudioWorkloadTest : oboe::AudioStreamDataCallback {
 
     /**
      * @brief Closes the audio stream.
-     * @return 0 on success, or a negative Oboe error code on failure.
-     */
+     * @return 0 on success, or a negative Oboe error code on failure. */
     int32_t close() {
-        if (mStream) {
-            oboe::Result result = mStream->close();
+        stop(); // Ensure stream is stopped before closing.
```

Comment:
Isn't close will always stop? What is more, there is not lock to guaranteed the stream is stopped after calling `stop`. `start` can always be called after calling `stop`

Reply:
Addressed all comments by adding a new lock that covers all functions that affect the stream